modules/security-aai/src/main/java/org/opencastproject/security/aai/DynamicLoginHandler.java

```diff
@@ -0,0 +1,432 @@
+    String displayName = extractDisplayName(request);
+    if (displayName != null && !"".equals(displayName)) {
+      return displayName;
+    }
+    List<String> givenNames = attributeMapper.getMappedAttributes(request, "givenName");
+    String givenName = "";
+    if (givenNames.size() > 0) {
+      String givenNameValue = givenNames.get(0);
+      givenName = StringUtils.isBlank(givenNameValue) ? ""
+              : new String(givenNameValue.getBytes(StandardCharsets.ISO_8859_1),
+                      StandardCharsets.UTF_8);
+    }
+    List<String> surnames = attributeMapper.getMappedAttributes(request, "sn");
+    String sn = "";
+    if (surnames.size() > 0) {
+      String surnameValue = surnames.get(0);
+      sn = StringUtils.isBlank(surnameValue) ? ""
+              : new String(surnameValue.getBytes(StandardCharsets.ISO_8859_1),
+                      StandardCharsets.UTF_8);
+    }
+    return StringUtils.join(new String[] { givenName, sn }, " ");
+  }
+
+  /**
+   * Extracts the e-mail from the request.
+   *
+   * @param request
+   *          the request
+   * @return the e-mail address
+   */
+  private String extractEmail(HttpServletRequest request) {
+    List<String> mailAdresses = attributeMapper.getMappedAttributes(request, "mail");
+    if (mailAdresses.size() > 0) {
+      String mailValue = mailAdresses.get(0);
+      String mail = StringUtils.isBlank(mailValue) ? ""
+              : new String(mailValue.getBytes(StandardCharsets.ISO_8859_1),
+                      StandardCharsets.UTF_8);
+      return mail;
+    } else {
+      return null;
+    }
+  }
+
+  /**
+   * Extracts the e-mail from the request.
+   *
+   * @param request
+   *          the request
+   * @return the e-mail address
+   */
+  private String extractDisplayName(HttpServletRequest request) {
+    List<String> displayNames = attributeMapper.getMappedAttributes(request, "displayName");
+    if (displayNames.size() > 0) {
+      String displayNameValue = displayNames.get(0);
+      String displayName = StringUtils.isBlank(displayNameValue) ? ""
+              : new String(displayNameValue.getBytes(StandardCharsets.ISO_8859_1),
+                      StandardCharsets.UTF_8);
+      return displayName;
+    } else {
+      return null;
+    }
+  }
+
+  /**
+   * Extracts the roles from the request.
+   *
+   * @param request
+   *          the request
+   * @return the roles
+   */
+  private Set<JpaRole> extractRoles(String id, HttpServletRequest request) {
+    List<String> aaiRoles = attributeMapper.getMappedAttributes(request, "roles");
+    JpaOrganization organization = fromOrganization(securityService.getOrganization());
+    Set<JpaRole> roles = new HashSet<JpaRole>();
+    if (aaiRoles != null) {
+      for (String aaiRole : aaiRoles) {
+        roles.add(new JpaRole(aaiRole, organization));
+      }
+    }
+    roles.add(new JpaRole(organization.getAnonymousRole(), organization));
+    if (StringUtils.equals(id, bootstrapUserId)) {
```

Comment:
 @jchssystems bootstrapUserId is loaded from the configuration file in the original ConfigurableLoginHandler. As you don't it in DynamicLoginHandler, bootstrapUserId will always be null. But since you can configure a bootstrap user in the security configuration, it is safe to just omit those lines here.